ledger/sandbox/src/main/scala/com/digitalasset/platform/indexer/RecoveringIndexer.scala

```diff
@@ -21,14 +21,14 @@ import scala.util.{Failure, Success}
   * @param restartDelay Time to wait before restarting the indexer after a failure
   * @param asyncTolerance Time to wait for asynchronous operations to complete
   */
-class RecoveringIndexer(
+final class RecoveringIndexer(
     scheduler: Scheduler,
     restartDelay: FiniteDuration,
     asyncTolerance: FiniteDuration,
-    loggerFactory: NamedLoggerFactory,
-) extends AutoCloseable {
+)(implicit ctx: LoggingContext)
+    extends AutoCloseable {
   private implicit val executionContext: ExecutionContext = DirectExecutionContext
-  private val logger = loggerFactory.getLogger(this.getClass)
+  private val logger = ContextualizedLogger.get[RecoveringIndexer]
```

Comment:
Can you use `this.type` in this sort of situation? That should help prevent copy-pasta 🍝 errors.

Reply:
Totally, I'll make this change, thanks. :)

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4097/commits/8445306ca57324207496849134905b1ecc6c5f03